File: src/main/java/org/bipmed/brave/server/query/Query.java
package org.bipmed.brave.server.query;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Query {

    private String snpId;
    private String assemblyId;
    private String datasetId;
    private String referenceName;
    private Long start;
    private Long end;
    private String geneSymbol;

}

File: src/main/kotlin/org/bipmed/brave/server/query/Query.kt
package org.bipmed.brave.server.query

import org.springframework.data.mongodb.core.mapping.Document

@Document
data class Query(
        val snpId: String? = null,
        val assemblyId: String? = null,
        val datasetId: String? = null,
        val referenceName: String? = null,
        val start: Long? = null,
        val end: Long? = null,
        val geneSymbol: String? = null
)